### PR TITLE
Team/5072 change release process

### LINKED ISSUE #5072: Change the way we release (Change request)
During the time @faboweb was out due to _force majeure_ Jordan and I discovered that the release process is somehow overcomplicated and somehow unpredictable.

Being develop the base branch for the Simsala branches, you never know for certain if it will then successfully merge to master after merging the Simsala PR, because it is not possible to see the conflicts.

Another issue with the current release process is that there too many steps:

We need to push to release (1) to open the Simsala branches (2), to merge these to develop (3) to then finally merge develop to master (4) and then (only if CHANGELOG.md changed) finally in fact release. This seems overcomplicated. It seems more reasonable to work with PRs from develop to master and then manually increment the version and CHANGELOG.md. I am not sure if this is the only thing Simsala actually does. Or another option would be to just work with release, develop and master. When doing `yarn release` in develop, the `release` branch opens a PR with base master and in that PR we automatically increment the version number & modify CHANGELOG.md.

In another PR (#5020), I also proposed staying the same but adding a merge strategy from develop to master so conflicts don't spoil our release process and we don't need to always make sure if something got released or not.

In any case, our release process seems quite cumbersome and is really taking us so much time, unfortunately

### PR DESCRIPTION
**Pull request by Bitcoinera**
Closes #5072 

**Description:**

For future successful releases! :champagne: 

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

File: .github/workflows/app_deploy_develop.yml [hunk 1]
```diff
@@ -36,7 +36,7 @@ jobs:
         id: deploy
         with:
           alias: staging
-          publish-dir: './app/dist'
+          deploy-dir: './app/dist'
```
Comment (faboweb) [inline .github/workflows/app_deploy_develop.yml:39]: is this correct?

File: .github/workflows/app_deploy_production.yml [hunk 2]
```diff
@@ -36,7 +36,7 @@ jobs:
         id: deploy
         with:
           production-branch: master
-          publish-dir: './app/dist'
+          deploy-dir: './app/dist'
```
Comment (faboweb) [inline .github/workflows/app_deploy_production.yml:39]: same

File: .github/workflows/app_preview_e2e.yml [hunk 3]
```diff
@@ -30,7 +30,7 @@ jobs:
         uses: nwtgck/actions-netlify@v1.1.10
         id: deploy
         with:
-          publish-dir: './app/dist'
+          deploy-dir: './app/dist'
```
Comment (faboweb) [inline .github/workflows/app_preview_e2e.yml:33]: same

File: .github/workflows/app_deploy_production.yml (hunk 2 above)
Comment (Bitcoinera) [reply .github/workflows/app_deploy_production.yml:39]: ah right...

Comment (Bitcoinera) [reply .github/workflows/app_deploy_production.yml:39]: yeah, it is wrong: https://github.com/nwtgck/actions-netlify

I went too far with the renaming :sweat_smile:

**Comment by faboweb** [comment]
🚀 Deployed on https://5f889ff80fa7ac1f419c1aa2--lunieio.netlify.app

**Review by faboweb — APPROVED** [review]

**Pull request merged**